packages/passport/sdk-sample-app/src/components/zkevm/EthSendTransactionExamples/SeaportFulfillAvailableAdvancedOrders.tsx

```diff
@@ -0,0 +1,172 @@
+
+function SeaportFulfillAvailableAdvancedOrders({ disabled, handleExampleSubmitted }: RequestExampleProps) {
+  const { orderbookClient } = useImmutableProvider();
+  const { zkEvmProvider } = usePassportProvider();
+
+  const [listingIds, setListingIds] = useState<string>('');
+  const [walletAddress, setWalletAddress] = useState<string>('');
+  const [isBuldingTransaction, setIsBuildingTransaction] = useState<boolean>(false);
+  const [transaction, setTransaction] = useState<PopulatedTransaction>();
+  const [transactionError, setTransactionError] = useState<string>('');
+
+  const seaportContractAddress = useMemo(
+    () => (
+      orderbookClient.config().seaportContractAddress
+    ),
+    [orderbookClient],
+  );
+
+  useEffect(() => {
+    const getAddress = async () => {
+      if (zkEvmProvider) {
+        const [address] = await zkEvmProvider.request({
+          method: 'eth_requestAccounts',
+        });
+        setWalletAddress(address || '');
+      }
+    };
+
+    getAddress().catch(console.log);
+  }, [zkEvmProvider, setWalletAddress]);
+
+  useEffect(() => {
+    setTransactionError('');
+    setTransaction(undefined);
+  }, [listingIds]);
+
+  const validate = useCallback(async () => {
+    setTransactionError('');
+    setTransaction(undefined);
+    setIsBuildingTransaction(true);
+
+    try {
+      const fulfillResponse = await orderbookClient.fulfillBulkOrders(
+        listingIds.replace(' ', '').split(',').map((orderId) => ({
+          listingId: orderId,
+          takerFees: [],
+        })),
+        walletAddress,
+      );
+
+      if (!fulfillResponse.sufficientBalance) {
+        setTransactionError('Insufficient balance to fulfill the order(s)');
+        return;
+      }
+
+      if (fulfillResponse.sufficientBalance) {
+        const {
+          actions, unfulfillableOrders,
+        } = fulfillResponse;
+
+        if (unfulfillableOrders.length > 0) {
+          setTransactionError('Not all orders are fulfillable');
+          return;
+        }
+
+        const verifyAction = actions.find((action) => (
+          action.type === ActionType.TRANSACTION && action.purpose === TransactionPurpose.APPROVAL
+        )) as TransactionAction | undefined;
+
+        if (verifyAction) {
+          const approvalTransaction = await verifyAction.buildTransaction();
+          await zkEvmProvider?.request({
+            method: 'eth_requestAccounts',
+            params: [approvalTransaction],
+          });
+        }
+
+        const transactionAction = actions.find((action) => (
+          action.type === ActionType.TRANSACTION && action.purpose === TransactionPurpose.FULFILL_ORDER
+        )) as TransactionAction | undefined;
+
+        if (!transactionAction) {
+          setTransactionError('Failed to find transaction to process');
+          return;
+        }
+
+        setTransaction(await transactionAction.buildTransaction());
```

Comment:
Will the actions contain more than one 
`action?type === ActionType.TRANSACTION && action.purpose === TransactionPurpose.FULFILL_ORDER` action? 
If so, why do we only perform first one?

Reply:
This will only contain more if the bulk buy limit with the orderbook is exceeded but otherwise bulk buy is just 1 transaction to sign. This is sufficient for sample-app testing purposes to be able to test a bulk buy transaction.